File: internal/model/book.go
package model

import "time"

type Book struct {
	ID            uint      `json:"id" gorm:"primaryKey"`
	UserID        uint      `json:"user_id" gorm:"index;not null"`
	BookMD5       string    `json:"book_md5" gorm:"size:32;index"`
	Title         string    `json:"title" gorm:"size:255;not null"`
	TotalChapters int       `json:"total_chapters" gorm:"not null"`
	CreatedAt     time.Time `json:"created_at" gorm:"autoCreateTime"`
}

File: android/app/src/main/java/com/storytrim/app/data/model/Book.kt
package com.storytrim.app.data.model

data class Book(
    val id: Long,
    val cloudId: Long,
    val userId: Long,
    val bookMd5: String?,
    val title: String,
    val totalChapters: Int,
    val createdAt: Long,
    val syncState: Int = 0 // 0: Local, 1: Synced, 2: CloudOnly
)

data class Chapter(
    val id: Long,
    val bookId: Long,
    val cloudId: Long,
    val index: Int,
    val title: String,
    val md5: String,
    val content: String? = null
)